cityofaustin/atd-vz-data
# Add location_notes table schema and trigger to LDM

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/13708

This PR adds migrations for the `location_notes` table schema (no data) and the one trigger set on it.

## Testing
**URL to test:** 
Local

**Steps to test:**
1. Make sure that your `.env` is updated with the `LDM_` variables shown in the `env_template`. Check the 1Pass secrets called `RDS Instance: LDM staging` for the values. `LDM_DATABASE` is `atd_vz_data`.
2. Replicate the LDM staging database locally by running:
```bash
$ ./vision-zero -l replicate-db
```
3. From the `atd-vzd` folder run:
```bash
$ hasura migrate apply --envfile .env.local
$ hasura metadata apply --envfile .env.local
```
4. To backfill the location notes data from the VZ read replica to your local LDM database, run:
```bash
export RR_PASSWORD=<your VZ read replica password>
export RR_USER=<your VZ read replica username>

docker exec -i \
-e PGHOST=db-rr.vision-zero.austinmobility.io \
-e PGDATABASE=atd_vz_data \
-e PGPASSWORD=$RR_PASSWORD \
-e PGUSER=$RR_USER \
visionzero-postgis \
pg_dump -t location_notes --data-only | \
perl -pe 's/^(SELECT pg_catalog\.set_config)/-- $1/g' | \
docker exec -i \
-e PGHOST=localhost \
-e PGDATABASE=atd_vz_data \
-e PGPASSWORD=visionzero \
-e PGUSER=visionzero \
visionzero-postgis \
psql --set ON_ERROR_STOP=on;
``` 
5. Through your database explorer or the local Hasura console, look for the `public.location_notes` table
6. Update the value of a column like `text` of a row in `public.location_notes`, and you should see the `updated_at` column update.
---
#### Ship list
